src/utils.ts

```diff
@@ -130,29 +156,65 @@ export function matchElementThatNavigateOnMouseEvent(params: {
       continue
     if (reaction.action.type !== 'NODE')
       continue
-    if (reaction.action.navigation !== 'NAVIGATE')
-      continue
     if (!reaction.action.destinationId)
       continue
 
-    const navigationNode: InteractiveNode = {
-      node,
-      parentFrameId: parentFrame.id,
+    const navigationNodeCommonProperties: ReactionDataCommonProperties & ReactionDataTriggerProperties = {
       triggerType: reaction.trigger.type,
-      destinationFrameId: reaction.action.destinationId,
-      generatedName: isGroup(node) ? generateGroupName(node) : node.name,
+      generatedName: generateNodeName(node),
     }
 
+    // TODO: tell the user in case the delay is set to 0
     if (reaction.trigger.delay > 0) {
       // In the Figma UI, the delay can be set only if the device is mobile and the events are
       // MOUSE_LEAVE, MOUSE_ENTER, TOUCH_DOWN, TOUCH_UP even if the TOUCH events are typed as mouse
       // ones. It's better to specify this detail in the docs
-      navigationNode.delay = reaction.trigger.delay * 1000
+      navigationNodeCommonProperties.delay = reaction.trigger.delay * 1000
     }
 
-    mutableInteractiveNodes.push(navigationNode)
-
-    // Can't break because the same node can have multiple mouse reactions
-    // break
+    if (reaction.action.navigation === 'NAVIGATE') {
+      result.push({
+        ...navigationNodeCommonProperties,
+        navigationType: reaction.action.navigation,
+        destinationFrameId: reaction.action.destinationId,
+        destinationFrameName: generateNodeName(figma.getNodeById(reaction.action.destinationId)!),
+      })
+      // Can't break because the same node can have multiple mouse reactions
+    }
+    if (reaction.action.navigation === 'SCROLL_TO') {
+      result.push({
+        ...navigationNodeCommonProperties,
+        navigationType: reaction.action.navigation,
+        destinationNodeId: reaction.action.destinationId,
+        destinationNodeName: generateNodeName(figma.getNodeById(reaction.action.destinationId)!),
+
+      })
+      // Can't break because the same node can have multiple mouse reactions
+    }
   }
+
+  return result
+}
+
+export function findParentRootFrame(node: BaseNode) {
+  const parent = node.parent
+
+  assertIsDefined(parent, `Parentless nodes are not expected (node id: ${node.id})`)
+
+  if (parent.type === 'FRAME' && isRootFrame(parent))
+    return parent
+
+  return findParentRootFrame(parent)
+}
+
+export function isRootFrame(node: FrameNode) {
+  if (node.parent?.type === 'FRAME')
+    return false
+
+  return true
+}
+
+export function assertIsDefined<T>(value: T, errorMessage: string): asserts value is NonNullable<T> {
```

Comment:
isn't assert enough and generic for this case? I'm just considering future uses for it, instead of a specific one.

Reply:
Yes but `assert` is not available in the browser, am I mistaken?

Reply:
Yes, I just meant a copy-paste implementation of it.

Reply:
I will check if I can use the Node.js one
https://github.com/nodejs/node/blob/v21.4.0/lib/assert.js
I must check the types of the module etc
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/assert/strict.d.ts

Reply:
In the end, I will opt to keep the original implementation because it's the only one where TypeScript can infer exactly what I need (values not being null or undefined).

For more context, here is part of out conversation

--- 

Uhm I' looking at the Node.js APIs
https://nodejs.org/api/assert.html#assertvalue-message
and to the types of the library you shared with me
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/assert/index.d.ts
but I can't see something like I need, am i mistaken?

what I'm looking for is something avoids me writing this
```js
const rootFrame = simplifiedFramesById[rootFrameId]
if(!rootFrame) throw new Error(`Root Frame not found (node id ${rootFrameId})`)
// rest of the code
 ```

and so I can write
```js
const rootFrame = simplifiedFramesById[rootFrameId]
assertIsDefined(rootFrame, `Root Frame not found (node id ${rootFrameId})`)
```
(now that I look at it, it does not look very different 😄 )

Maybe I should change the API to write something like
```js
const rootFrame = simplifiedFramesById[rootFrameId]
assertIsDefined(rootFrame)
```
but the point is that I need TS support. I mean, TS must infer that if the value is not defined, the program explodes, and viceversa.